### ISSUE TITLE
Tree performance with many childrens expanded

### ISSUE THREAD

**Issue by ghost** [oria/gridx#290]
Hi GridX Team,
I notice significant performance differences between the "GridX playground" and my code.
This is my grid config:

// GridX 1.3.3

layout = [
    { id: "name", field: "name", name:"Name"},
    { id: "age", field: "age", name:"Age"},
    { id: "code", field: "code", name:"Code"}
];

data = [];
for(var i=0; i<1000; i++)
{
    data.push({kid: "item-"+i, name: "Name" + i, age: 0, code: "C+" + i, 
        ch: [
            {kid: 5000 + "item-"+i, name: "CH1-Name" + i, age: 0, code: "CH1-C+" + i, 
                ch: [{kid: 9000 + "item-"+i, name: "CH1-CH1-Name" + i}]
            },
            {kid: 6000 + "item-"+i, name: "CH2-Name" + i, age: 0, code: "CH2+" + i},
            {kid: 7000 + "item-"+i, name: "CH3-Name" + i, age: 0, code: "CH3+" + i}
        ]
    });
}
var newStore = function(dataSource){
    var store = new Memory({
        data: dataSource,
        idProperty: 'id',
        chProperty: 'ch'
    });
    store.hasChildren = function(id, item)
    {
        return item.ch && item.ch.length;
    };
    store.getChildren = function(item)
    {
        return item.ch;
    };
    return store;
};
dataStore = newStore(data);

var config = {
        id: "gridx1",
        cacheClass: cacheSync,
        store: dataStore,
        structure: layout,
        modules: [
            Tree,
            VirtualVScroller
        ],
        vScrollerLazy: true
    };

g = new Grid(config);
g.startup();
g.placeAt("gridx_cnt");
g.body.refresh();

// multiple expand
for(var i=0; i<500; i++){
    g.tree.expand("item-"+i, true);
}
g.body.refresh();

Now, if i try to scroll or expand\collapse manually one row, it takes a long time.
But, if i comment this code in modules/View.js(absent in 'GridX playground'), is responsive:
item.openned.sort(function(a, b){
    return m.idToIndex(a) - m.idToIndex(b);
});

BUT, there is alternative code, or gridx update, to resolve this issue?
Thank you.

**Comment by slart23** [comment]
I am experiencing this same problem, and exactly as described in the original post, although I am not using vScrollerLazy:true.  The browser is completely unresponsive when scrolling a grid containing many expanded rows, until all processing of the scroll events has completed.  I have pinpointed that same code, the sorting of expanded (open) items, as the primary source by far of the lengthy execution time.

At 500+ top level rows, each with 10 child rows, each of these sorts can take up to 50 ms.  Given the apparent synchronous execution of all code involved in processing the scroll events, many of these sorts can add up to several seconds before control of the event queue is finally released to the browser to show any responsiveness.

I would like to know if a fix can be made to what is for me a significant performance issue.

Thanks.

**Comment by slart23** [comment]
Hi, wondering if anyone else has also observed this issue, or may even have suggestions for a fix or workaround?  Thanks.

**Comment by slart23** [comment]
Checking in again to see if anyone else has observed this issue, or better yet has advice.  This frequently called sort can leave the browser unresponsive for some time when scrolling a grid with many expanded rows.  I haven't found a way around it to date.

**Comment by shanghaikid** [comment]
yes, they have big performance issue if you have too many children. and they don't provide server-side tree solution neither. 
Maybe you can write your own body/view/vScroller modules, too many dom operations in their body/view and the virtual vScroller isn't good enough. Good luck.
